typegen/strats/inline.py

```diff
@@ -0,0 +1,262 @@
+
+        if rettypes.shape[0] == 1:
+            rettype = rettypes[TraceData.VARTYPE].values[0]
+            assert rettype is not None
+
+            ret_hint = rettype.__name__
+            logger.debug(f"Applying return type hint '{ret_hint}' to '{fdef.name}'")
+            fdef.returns = ast.Name(ret_hint)
+
+    def _find_class(self, node: ast.AST) -> str | None:
+        class_name = None
+        while hasattr(node, "parent"):
+            node = node.parent  # type: ignore
+            if isinstance(node, ast.ClassDef):
+                class_name = node.name
+                break
+
+        return class_name
+
+    def _find_targets(self, node: ast.Assign | ast.AnnAssign | ast.AugAssign):
+        if isinstance(node, ast.AnnAssign | ast.AugAssign):
+            target = node.target
+            return self._extract_target_names_with_nodes(target)
+        else:
+            targets = node.targets
+            names = list()
+            for target in targets:
+                names += self._extract_target_names_with_nodes(target)
+            return names
+
+    def visit_AugAssign(self, node: ast.AugAssign) -> ast.AST | list[ast.AST]:
+        replace = self._visit_assigns(node)
+        assert isinstance(replace, list)
+
+        return replace
+
+    def visit_Assign(self, node: ast.Assign) -> ast.AST | list[ast.AST]:
+        return self._visit_assigns(node)
+
+    def _visit_assigns(
+        self, node: ast.Assign | ast.AugAssign
+    ) -> ast.AST | list[ast.AST]:
+        if not node.value:
+            return node
+
+        target_names_with_nodes = self._find_targets(node)
+        target_names = [
+            target_name_with_node[0]
+            for target_name_with_node in target_names_with_nodes
+        ]
+
+        logger.debug(f"Applying hints to '{target_names}'")
+
+        class_name = self._find_class(node)
+        if class_name is not None:
+            names = self.df[TraceData.CLASS].map(lambda t: t.__name__ if t else None)
+            class_check = names == class_name
+        else:
+            class_check = self.df[TraceData.CLASS].isnull()
+
+        var_mask_local_variables = [
+            self.df[TraceData.CATEGORY] == TraceDataCategory.LOCAL_VARIABLE,
+            self.df[TraceData.VARNAME].isin(target_names),
+            self.df[TraceData.LINENO] == node.lineno,
+            class_check,
+        ]
+
+        var_mask_class_members = [
+            self.df[TraceData.CATEGORY] == TraceDataCategory.CLASS_MEMBER,
+            self.df[TraceData.VARNAME].isin(target_names),
+            class_check,
+        ]
+
+        local_traced_vars = self.df[
+            functools.reduce(operator.and_, var_mask_local_variables)
+        ]
+        class_members = self.df[functools.reduce(operator.and_, var_mask_class_members)]
+
+        # Attach hint directly to assignment and promote to AnnAssign
+        new_nodes: list[ast.AST] = []
+        contains_one_target = len(target_names_with_nodes) == 1
```

Comment:
Add "and not isinstance(node, ast.AugAssign)" and change the name to only_update_node.

Reply:
Done